### PR TITLE
CATROID-571 Refactor RepeatActionTest to Kotlin

### PR DESCRIPTION
**Pull request by trummerp**
### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catroid/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [x] Include the name of the Jira ticket in the PR’s title
- [ ] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [x] Verify to commit no other files than the intentionally changed ones
- [x] Include reasonable and readable tests verifying the added or changed behavior
- [x] Confirm that new and existing unit tests pass locally
- [x] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [x] Stick to the project’s gitflow workflow
- [x] Verify that your changes do not have any conflicts with the base branch
- [x] After the PR, verify that all CI checks have passed
- [ ] Post a message in the *catroid-stage* or *catroid-ide* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### CODE REVIEW

File: catroid/src/test/java/org/catrobat/catroid/test/content/actions/RepeatActionTest.kt [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+import org.junit.runners.JUnit4
+
+@RunWith(JUnit4::class)
+class RepeatActionTest {
+    private var sprite: Sprite? = null
+    private var script: Script? = null
+    private var repeatBrick: RepeatBrick? = null
+    private var changeYByNBrick: ChangeYByNBrick? = null
+    private var repeatTimesFormula: Formula? = null
+
```
Comment (Koell) [inline catroid/src/test/java/org/catrobat/catroid/test/content/actions/RepeatActionTest.kt:46]: please change all the var's to lateinit  and not nullable e.g.  "private lateinit var sprite: Sprite"
with this you don't need to force the variables to asume that they are not null

.!! without a null check is just hoping that there can't occur a nullpointer exception

File: catroid/src/test/java/org/catrobat/catroid/test/content/actions/RepeatActionTest.kt [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+import org.junit.Assert
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.Parameterized
+
+@RunWith(Parameterized::class)
+class RepeatActionTest(
+    private val name: String,
+    private val repeatTimes: Formula?,
```
Comment (ndrnour) [inline catroid/src/test/java/org/catrobat/catroid/test/content/actions/RepeatActionTest.kt:43]: please change `repeatTimes` parameter to Formula type, also just remove the `?` at the end

**Review by ndrnour — APPROVED** [review]

**Review by Koell — APPROVED** [review]
Thanks

**Review by wslany — CHANGES_REQUESTED** [review]
Thanks. However, the tests in Java were more comprehensive: they tested, e.g., whether the loop delay is correct. @trummerp why have these important and meaningful tests been silently dropped? And @Koell and @ndrnour why was this not flagged in the code review? 

Having said that, it is necessary to add a CLT for this brick, and then the more complex tests can be relegated there. If possible, please add such a CLT for the brick in https://github.com/Catrobat/Catroid/tree/develop/catroid/src/androidTest/assets/catrobatTests/data instead of adding them to the Kotlin test. And please also add the test cases from the Kotlin tests to the CLTs.

We want to use the CLTs as a formal, executable, but device independent specification in the form of automatic tests of the Catrobat language. Catty and other projects will be able to execute them, and any future implementation can use them as a reference against which to implement it.

**Comment by ndrnour** [comment]
> Thanks. However, the tests in Java were more comprehensive: they tested, e.g., whether the loop delay is correct. @trummerp why have these important and meaningful tests been silently dropped? And @Koell and @ndrnour why was this not flagged in the code review?

@wslany sorry for the late reply, I've just seen it. answering honestly, I don't spend much time reviewing such tickets cause they are meaningless to me!  we have many tickets that are more important than just refactoring some test cases from one language to another.

**Pull request closed without merging**